test_conformance/api/test_queries.cpp

```diff
@@ -131,86 +131,95 @@ int test_get_platform_info(cl_device_id deviceID, cl_context context, cl_command
     return 0;
 }
 
-int test_get_sampler_info(cl_device_id deviceID, cl_context context, cl_command_queue queue, int num_elements)
+template <typename T>
+int sampler_param_test(cl_sampler sampler, cl_sampler_info param_name,
+                       T expected, const char *name)
 {
-    int error;
     size_t size;
-
-    PASSIVE_REQUIRE_IMAGE_SUPPORT( deviceID )
-
-    cl_sampler_properties properties[] = {
-        CL_SAMPLER_NORMALIZED_COORDS, CL_TRUE,
-        CL_SAMPLER_ADDRESSING_MODE, CL_ADDRESS_CLAMP,
-        CL_SAMPLER_FILTER_MODE, CL_FILTER_LINEAR,
-        0 };
-    clSamplerWrapper sampler = clCreateSamplerWithProperties(context, properties, &error);
-    test_error( error, "Unable to create sampler to test with" );
-
-    cl_uint refCount;
-    error = clGetSamplerInfo( sampler, CL_SAMPLER_REFERENCE_COUNT, sizeof( refCount ), &refCount, &size );
-    test_error( error, "Unable to get sampler ref count" );
-    if( size != sizeof( refCount ) )
+    T val;
+    int error = clGetSamplerInfo(sampler, param_name, sizeof(val), &val, &size);
+    test_error(error, "Unable to get sampler info");
+    if (val != expected)
```

Comment:
This could use the `test_assert_error()` helper macro.

Reply:
In this case (and a few others below) this removes some of the failure information - it removes the 'name' of the parameter being tested.